packages/react-components/react-spinbutton/src/components/SpinButton/SpinButton.test.tsx

```diff
@@ -318,6 +346,97 @@ describe('SpinButton', () => {
+    const spinButton = getSpinButtonInput();
+
+    userEvent.click(decrementButton);
+    // Already at min bound, no change
+    expect(onChange).not.toHaveBeenCalled();
+    fireEvent.keyDown(spinButton, { key: Keys.ArrowDown });
+    expect(onChange).not.toHaveBeenCalled();
+    fireEvent.keyDown(spinButton, { key: Keys.PageDown });
+    expect(onChange).not.toHaveBeenCalled();
+
+    userEvent.click(incrementButton);
+    // From 0 --> 1
+    expect(onChange).toHaveBeenCalledTimes(1);
+
+    userEvent.click(incrementButton);
+    // At max bound, no change
+    expect(onChange).toHaveBeenCalledTimes(1);
+    fireEvent.keyDown(spinButton, { key: Keys.ArrowUp });
+    expect(onChange).toHaveBeenCalledTimes(1);
+    fireEvent.keyDown(spinButton, { key: Keys.PageUp });
+    expect(onChange).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not call change at bounds when controlled', () => {
+    const onChange = jest.fn();
+
+    const { getAllByRole, rerender } = render(<SpinButton value={0} min={0} max={1} onChange={onChange} />);
+
+    const [incrementButton, decrementButton] = getAllByRole('button');
+    const spinButton = getSpinButtonInput();
+
+    userEvent.click(decrementButton);
+    // Already at min bound, no change
+    expect(onChange).not.toHaveBeenCalled();
+    fireEvent.keyDown(spinButton, { key: Keys.ArrowDown });
+    expect(onChange).not.toHaveBeenCalled();
+    fireEvent.keyDown(spinButton, { key: Keys.PageDown });
+    expect(onChange).not.toHaveBeenCalled();
+
+    rerender(<SpinButton value={1} min={0} max={1} onChange={onChange} />);
+
+    userEvent.click(incrementButton);
+    // Already at maz bound, no change
+    expect(onChange).not.toHaveBeenCalled();
+    fireEvent.keyDown(spinButton, { key: Keys.ArrowUp });
+    expect(onChange).not.toHaveBeenCalled();
+    fireEvent.keyDown(spinButton, { key: Keys.PageUp });
+    expect(onChange).not.toHaveBeenCalled();
+  });
+
+  it('calls on change when defaultValue is `null` when uncontrolled', () => {
+    const onChange = jest.fn();
+    const { getAllByRole } = render(<SpinButton defaultValue={null} onChange={onChange} />);
+
+    const [incrementButton, decrementButton] = getAllByRole('button');
+    userEvent.click(incrementButton);
+
+    expect(onChange).toHaveBeenCalledTimes(1);
+    expect(getSpinButtonInput().value).toBe('1');
+
+    userEvent.click(decrementButton);
+
+    expect(onChange).toHaveBeenCalledTimes(2);
+    expect(getSpinButtonInput().value).toBe('0');
+  });
+
+  it('calls on change when value is `null` when controlled', () => {
+    const onChange = jest.fn();
+    const { getAllByRole, rerender } = render(<SpinButton value={null} min={5} max={10} onChange={onChange} />);
+
+    const [incrementButton, decrementButton] = getAllByRole('button');
+    userEvent.click(incrementButton);
+
+    expect(onChange).toHaveBeenCalledTimes(1);
+    expect(onChange.mock.calls[0][1]).toEqual({ value: 5, displayValue: undefined });
```

Comment:
Since the assumed starting value here is 5, shouldn't `onChange` be called with `value: 6` after the incrementButton click?

Reply:
I think you can make an argument for that. This behavior is based on the `input type="number"` which you can observe here: https://codepen.io/seanms/pen/BaYZVrL?editors=1010